### DISCUSSION TITLE
¿Cómo añadir argumentos al store.despatch?

### DISCUSSION THREAD

**Discussion by RaulRomero26** [LaunchX-InnovaccionVirtual/FrontEnd-Mision#489] [Q&A] [answered]
Mi problema es el siguiente, logro imprimir, ya componentes diferentes, sin embargo para la acción de los botones requiero, poder  mandar parametros hacia mi funcion de dispatch: 
![image](https://user-images.githubusercontent.com/55547165/159602166-e3da05c8-9a2d-4c92-80c4-7f9fb83f9cfe.png)
en mi store:
![image](https://user-images.githubusercontent.com/55547165/159602209-35805e24-6cbf-4ef4-8126-006792675a80.png)
si embargo al imprimir el "itemID" me sale undefined
![image](https://user-images.githubusercontent.com/55547165/159602312-ee0ad981-024f-406f-952f-ffcb70532a50.png)
mi componente ya recibe por props un item referente sobre el cual va  ser el contador, lo que no puedo es subir o bajar específicamente el contador o existencias de ese ítem en especial. Ya investigue y se supone que mandandole un objeto al dispatch debería funcionar pero lo lo logro hacer funcionar.
Mas sobre mi store: 
![image](https://user-images.githubusercontent.com/55547165/159602441-3fa322f6-5894-4354-aed2-efa9f61c34ce.png)

**Comment by doguedogue** [comment]
`subirContado( { commit },   itemID  ){ ...`

**Reply by RaulRomero26** [reply]
pero no es en el action del button, eso es ya en la funcion en el store, mi problema es que al accionar el boton mande ciertos parametros que ya tengo, se que existen por que si los imprime en el template, pero no los envia.

**Comment by raulquintero13** [comment, marked as answer]
Yo lo hice asi para mandar parametros con el $store.dispatch, si es que te entendi bien:

en el contenedor:
< div  :class="flavor.class" @click="$store.dispatch('addFlavor',{flavor})" >
          

y en  actions lo cacho asi

actions:{
addFlavor(state, payload){
      const flavor = payload.flavor;

**Reply by RaulRomero26** [reply]
te juro que intente eso y no me lo detecta o me manda el item.id como undefined, cosa que si que si tiene por que lo imrpimo en el template

**Reply by RaulRomero26** [reply]
No manches! el problema era, que el argumento tiene que llamarse payload, o en su defecto llamarlo igual que al desestructurar con eso ya funcionan los contadores independientemente, gracias!
